ta/arch/arm/user_ta_header.c

```diff
@@ -46,7 +48,21 @@ void __noreturn _C_FUNCTION(__ta_entry)(unsigned long func,
 					struct utee_params *up,
 					unsigned long cmd_id)
 {
-	TEE_Result res = __utee_entry(func, session_id, up, cmd_id);
+	TEE_Result res;
+
+#if defined(CFG_TA_STACK_PROTECTOR)
+	uintptr_t canary;
+
+	res = _utee_cryp_random_number_generate(&canary, sizeof(canary));
```

Comment:
Doing this each time we enter a TA might be a bit expensive.

Reply:
Good point. I added check to only set the stack canary once